code/modules/antagonists/heretic/heretic_antag.dm

```diff
@@ -235,8 +236,8 @@
 	return ..()
 
 /datum/antagonist/heretic/ui_status(mob/user, datum/ui_state/state)
-	if(user.stat == DEAD)
-		return UI_CLOSE
+	if(owner.current?.stat == DEAD)
+		return UI_UPDATE
```

Comment:
This changes existing behavior? The heretic isn't intended to have their UI open while dead.

Reply:
atomized it